spec/services/exchange_rates/upload_monthly_csv_spec.rb

```diff
@@ -0,0 +1,103 @@
+# rubocop:disable RSpec/MultipleMemoizedHelpers
+RSpec.describe ExchangeRates::UploadMonthlyCsv do
+  subject(:upload_csv) { described_class.call }
+
+  let(:current_time) { Time.zone.now }
```

Comment:
I wonder if we could just pass these to the service and then we can avoid all of the memoization

Reply:
To be honest i quite liked it being like an atomic action so its all held within the class. I did have at first passing in the date as this might help further down the line with other updates to the CSV file but for now we dont have all the full requirements.

Reply:
Theres some extra memoization to allow the `.for_month` to only expect two integers which could just accept a date i guess? but think nicer to work like this.

Reply:
Yeah. I think I just noticed the bypassing of the rubocop rules and figured something smells here with the lots of setup.

Reply:
Do you think this is a blocker? i know ive worked where very mixed opinions on this cop? All id be doing is moving around to another class sort of spreading dependency potentially?

Reply:
Or could just not store some string values and add them inline which seems a but pointless 🤷

Reply:
TBH, your approach is actually more readable for the things that are conceptually relevant to the test.